### PR TITLE
Pre release (code review needed)

### LINKED ISSUE #46: MIDITrackView does too much work in body
It creates an AppleSequencer and reads files. Need to create a `MIDITrackViewModel` to store that.

### PR DESCRIPTION
**Pull request by emurray2**
This commit fixes AudioKit/AudioKitUI#46 and AudioKit/AudioKitUI#47. Rewrote the MIDITrackView entirely in SwiftUI, used `drawingGroup()` to render one path layer for the notes to improve drawing efficiency and created MIDITrackViewModel to store note rectangles and track properties for drawing.

### CODE REVIEW

**Comment by emurray2** [comment]
Hi @aure @wtholliday @orchetect . Was wondering if any of you wanted to check this out. No rush, but I want to publish this soon somewhere so I can share it with the community.

File: Sources/MIDITrackView/MIDITrackView.swift [hunk 1]
```diff
@@ -2,6 +2,15 @@
 
 import SwiftUI
 
+struct MIDINoteView: Shape {
```
Comment (wtholliday) [inline Sources/MIDITrackView/MIDITrackView.swift]: Probably should be called `MIDINotesView` (plural)

File: Sources/MIDITrackView/MIDITrackView.swift [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
+    let midiNotes: [CGRect]
```
Comment (wtholliday) [inline Sources/MIDITrackView/MIDITrackView.swift]: Call this `noteRects`

File: MIDITrackView.playground/Contents.swift [hunk 3]
```diff
@@ -5,18 +5,16 @@ import SwiftUI
...
-            MIDINote(position: 80, level: 5, length: 102, height: 5),
-            MIDINote(position: 103, level: 23, length: 75, height: 5),
-            MIDINote(position: 157, level: 36, length: 25, height: 5),
-            MIDINote(position: 208, level: 5, length: 102, height: 5),
-            MIDINote(position: 301, level: 23, length: 75, height: 5),
-            MIDINote(position: 376, level: 36, length: 25, height: 5),
-            MIDINote(position: 402, level: 5, length: 102, height: 5),
-            MIDINote(position: 450, level: 23, length: 75, height: 5),
-            MIDINote(position: 500, level: 36, length: 25, height: 5)
+            CGRect(x: 0.5, y: 0.0, width: 1.0, height: 10.0),
```
Comment (wtholliday) [inline MIDITrackView.playground/Contents.swift:8]: Again, it's odd for something called `midiNotes` to be `CGRects`

**Review by wtholliday — CHANGES_REQUESTED** [review]

**Review by aure — APPROVED** [review]

**Pull request merged**